segmentstore/server/src/main/java/io/pravega/segmentstore/server/attributes/SegmentAttributeBTreeIndex.java

```diff
@@ -671,29 +704,47 @@ private void storeInCache(List<Map.Entry<Long, ByteArraySegment>> toAdd, Collect
 
             // Add new ones.
             for (val e : toAdd) {
-                long offset = e.getKey();
-                ByteArraySegment data = e.getValue();
-                CacheEntry entry = this.cacheEntries.getOrDefault(offset, null);
-                if (entry == null || entry.getSize() != data.getLength()) {
-                    entry = new CacheEntry(offset, data.getLength(), this.currentCacheGeneration);
-                    this.cacheEntries.put(offset, entry);
-                }
-
-                storeInCache(entry, data);
+                storeInCache(e.getKey(), e.getValue());
             }
         }
     }
 
     @GuardedBy("cacheEntries")
-    private void storeInCache(CacheEntry entry, ByteArraySegment data) {
-        int newAddress;
+    private void storeInCache(long entryOffset, ByteArraySegment data) {
+        CacheEntry entry = this.cacheEntries.getOrDefault(entryOffset, null);
+        if (entry != null && entry.getSize() == data.getLength()) {
+            // Already cached.
+            return;
+        }
+
+        // Insert/Update the cache. There is a chance this won't go through, so be able to handle that. It's not the end
+        // of the world if we can't update the cache (as we always have the data in the attribute segment). If we are
+        // having trouble, log the error and move on.
+        int entryAddress = entry == null ? CacheEntry.NO_ADDRESS : entry.getCacheAddress();
+        try {
+            if (entry != null && entry.isStored()) {
+                // Entry exists. Replace it.
+                entryAddress = this.cacheStorage.replace(entry.getCacheAddress(), data);
```

Comment:
Codecov complains this line is not covered.
Seems like an important case to cover.

Reply:
See explanation above.